packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmSetTests.kt

```diff
@@ -740,6 +764,68 @@ internal abstract class ManagedSetTester<T>(
         }
     }
 
+    override fun remove() {
+        // TODO https://github.com/realm/realm-kotlin/issues/1097
+        //  Ignore RealmObject: structural equality cannot be assessed for this type when removing
+        //  elements from the set
+        if (classifier != RealmObject::class) {
+            val dataSet = typeSafetyManager.dataSetToLoad
+
+            errorCatcher {
+                realm.writeBlocking {
+                    val set = typeSafetyManager.createContainerAndGetCollection(this)
+                    set.add(dataSet[0])
+                    assertTrue(set.remove(dataSet[0]))
+                    assertTrue(set.isEmpty())
+                }
+            }
+
+            assertContainerAndCleanup { container ->
+                val set = typeSafetyManager.getCollection(container)
+                assertTrue(set.isEmpty())
+            }
+        }
+    }
+
+    override fun removeAll() {
+        // TODO https://github.com/realm/realm-kotlin/issues/1097
+        //  Ignore RealmObject: structural equality cannot be assessed for this type when removing
+        //  elements from the set
+        if (classifier != RealmObject::class) {
+            val dataSet = typeSafetyManager.dataSetToLoad
+
+            errorCatcher {
+                realm.writeBlocking {
+                    val set = typeSafetyManager.createContainerAndGetCollection(this)
+                    set.addAll(dataSet)
+                    assertTrue(set.removeAll(dataSet))
+
+                    // TODO https://github.com/realm/realm-kotlin/issues/1097
+                    //  If the RealmAny instance contains an object it will NOT be removed until
+                    //  the issue above is fixed
```

Comment:
Shouldn't this be tracked as separate issue? One thing is that our `RealmObject.equals` has the wrong semantics, but it is a more critical issue that we are not removing objects just because the current implementation relies on that `equals`. We should probably prioritize the actual issue of removal of `RealmAny` containing a `RealmObject` higher ... and I guess on fix could just be to fix the `equals`.

Reply:
That's true, the comment is misleading considering the issue doesn't mention `RealmAny`. I will modify the issue to also consider `RealmAny` containing objects as well. I agree we should do this rather soon as there seem to be people interested in it.